packages/react-email/src/cli/utils/preview/hot-reloading/create-dependency-graph.ts

```diff
@@ -0,0 +1,244 @@
+          */
+          if (!isJavascriptModule(pathToDependencyFromDirectory)) {
+            pathToDependencyFromDirectory = `${pathToDependencyFromDirectory}${path.extname(
+              filePath,
+            )}`;
+          }
+
+          return pathToDependencyFromDirectory;
+        } else {
+          // when either the path is a module or is absolute
+          return dependencyPath;
+        }
+      },
+    );
+
+    const moduleDependencies = importedPathsRelativeToDirectory.filter(
+      (dependencyPath) =>
+        !dependencyPath.startsWith('.') && !path.isAbsolute(dependencyPath),
+    );
+
+    const nonNodeModuleImportPathsRelativeToDirectory =
+      importedPathsRelativeToDirectory.filter(
+        (dependencyPath) =>
+          dependencyPath.startsWith('.') || path.isAbsolute(dependencyPath),
+      );
+
+    return {
+      dependencyPaths: nonNodeModuleImportPathsRelativeToDirectory,
+      moduleDependencies,
+    };
+  };
+
+  const updateModuleDependenciesInGraph = async (moduleFilePath: string) => {
+    const module = graph[moduleFilePath] ?? {
+      path: moduleFilePath,
+      dependencyPaths: [],
+      dependentPaths: [],
+      moduleDependencies: [],
+    };
+
+    const { moduleDependencies, dependencyPaths: newDependencyPaths } =
+      await getDependencyPaths(moduleFilePath);
+
+    module.moduleDependencies = moduleDependencies;
+
+    // we go through these to remove the ones that don't exist anymore
+    for (const dependencyPath of module.dependencyPaths) {
+      // Looping through only the ones that were on the dependencyPaths but are not
+      // in the newDependencyPaths
+      if (newDependencyPaths.includes(dependencyPath)) continue;
+
+      const dependencyModule = graph[dependencyPath];
+      if (dependencyModule !== undefined) {
+        dependencyModule.dependentPaths =
+          dependencyModule.dependentPaths.filter(
+            (dependentPath) => dependentPath !== moduleFilePath,
+          );
+      }
+    }
+
+    module.dependencyPaths = newDependencyPaths;
+
+    for (const dependencyPath of newDependencyPaths) {
+      const dependencyModule = graph[dependencyPath];
+      if (
+        dependencyModule !== undefined &&
+        !dependencyModule.dependentPaths.includes(moduleFilePath)
+      ) {
+        dependencyModule.dependentPaths.push(moduleFilePath);
+      } else {
+        /*
+          This import path might have not been initialized as it can be outside
+          of the original directory we looked into.
+        */
+        graph[dependencyPath] = {
+          path: dependencyPath,
+          moduleDependencies: [],
+          dependencyPaths: [],
+          dependentPaths: [moduleFilePath],
+        };
+      }
+    }
+
+    graph[moduleFilePath] = module;
+  };
+
+  for (const filePath of modulePaths) {
+    await updateModuleDependenciesInGraph(filePath);
+  }
+
+  const removeModuleFromGraph = (filePath: string) => {
```

Comment:
This function only deleted the node from the graph, and then deletes itself out of the `dependentPaths` of all other nodes it can find. It doesn't delete itself off of the dependencies of other nodes because this would end up causing unwanted behavior since this will happen anyway once the user changes removing the import from the dependencies.